snackpub/Dockerfile

```diff
@@ -0,0 +1,63 @@
+ARG node_version
+
+# Set up dev image
+FROM node:${node_version}-alpine AS dev
+WORKDIR /app
+
+RUN apk add --no-cache --upgrade \
+    bash \
+    busybox \
+    g++ \
+    make \
+    python3
+
+# Set up monorepo
+COPY . ./
+# Install dependencies
+RUN yarn install --frozen-lockfile
+# Build monorepo
+RUN yarn build
+
+# Start snackpub
+WORKDIR /app/snackpub
+CMD ["yarn", "start"]
+
+# Set up production image
+FROM node:${node_version}-alpine AS run
+WORKDIR /app
+
+RUN apk add --no-cache --upgrade \
+    bash \
+    busybox \
+    g++ \
+    make \
+    python3
+
+# - root monorepo files
+COPY --from=dev /app/package.json ./
+COPY --from=dev /app/yarn.lock ./
+# - snackpub files
+COPY --from=dev /app/snackpub/src ./snackpub/src
+COPY --from=dev /app/snackpub/package.json ./snackpub/package.json
+COPY --from=dev /app/snackpub/.env-cmdrc.json ./snackpub/.env-cmdrc.json
+COPY --from=dev /app/snackpub/tsconfig.json ./snackpub/tsconfig.json
+COPY --from=dev /app/snackpub/resources/redislabs_ca.pem ./snackpub/resources/redislabs_ca.pem
+
+# Set up envs
+ENV NODE_ENV "production"
+
+ARG DEPLOY_ENVIRONMENT
+ENV DEPLOY_ENVIRONMENT ${DEPLOY_ENVIRONMENT}
+
+# Reinstall dependencies to link the monorepo
+RUN yarn install --frozen-lockfile --production=false
+
+WORKDIR /app/snackpub
+# Build snackpub
+RUN yarn build
+# Reinstall only production dependencies
+RUN yarn install --frozen-lockfile --production
+# Finalize production image
+RUN yarn cache clean
+
+CMD ["node", "."]
```

Comment:
Maybe it would be nice to create a "production" stage before finalizing this image, to avoid having installed unnecessary system libraries in production. (e.g. like [Snackager](https://github.com/expo/snack/blob/main/snackager/Dockerfile#L47-L69))

```suggestion
# Set up envs
ENV NODE_ENV "production"

ARG DEPLOY_ENVIRONMENT
ENV DEPLOY_ENVIRONMENT ${DEPLOY_ENVIRONMENT}

# Reinstall dependencies to link the monorepo
RUN yarn install --frozen-lockfile --production=false

WORKDIR /app/snackpub
# Build snackpub
RUN yarn build
# Reinstall only production dependencies
RUN yarn install --frozen-lockfile --production
# Finalize production image
RUN yarn cache clean

# Build minimized production image
FROM node:${node_version}-alpine

# Set up envs
ENV NODE_ENV "production"

ARG DEPLOY_ENVIRONMENT
ENV DEPLOY_ENVIRONMENT ${DEPLOY_ENVIRONMENT}

COPY --from=dev /app/snackpub ./snackpub

WORKDIR /app/snackpub

CMD ["node", "."]
```